endpoint-common/src/error.rs

```diff
@@ -91,3 +92,30 @@ impl From<EndpointError> for HttpEndpointError {
         HttpEndpointError(err)
     }
 }
+
+#[derive(Debug)]
+pub struct CoapEndpointError(pub EndpointError);
+
+impl CoapEndpointError {
```

Comment:
I wonder if this should be moved to coap-endpoint crate, then you can remove endpoint-common's dependency on coap-lite as well.

Reply:
It originally was in the coap-endpoint crate. But looking at the http-endpoint error, I shifted it to endpoint-common. 

No issues, I'll move it back since now that I see, the mqtt-endpoint has it's error impl in its own crate.